Assets/MixedRealityToolkit.Services/InputSystem/FocusProvider.cs

```diff
@@ -854,70 +840,30 @@ private bool RaycastGraphicsStep(PointerData pointerData, RayStep step, LayerMas
         /// </summary>
         private void UpdateFocusedObjects()
         {
-            Debug.Assert(pendingPointerSpecificFocusChange.Count == 0);
-            Debug.Assert(pendingOverallFocusExitSet.Count == 0);
-            Debug.Assert(pendingOverallFocusEnterSet.Count == 0);
-
-            // NOTE: We compute the set of events to send before sending the first event
-            //       just in case someone responds to the event by adding/removing a
-            //       pointer which would change the structures we're iterating over.
```

Comment:
Is this no longer a valid concern?

Reply:
Actually not. Proceeding with your recommendation this now just counts pointers for each focusable gameobject.